src/chatdbg/util/text.py

```diff
@@ -16,7 +16,7 @@ def strip_ansi(s: str) -> str:
     return ansi_escape.sub("", s)
 
 
-def truncate_proportionally(text, maxlen=32000, top_proportion=0.5):
+def truncate_proportionally(text: str, maxlen: int=32000, top_proportion: Union[int, float]=0.5) -> str:
```

Comment:
`top_proportion` probably should just be `float`. Always replace `Union[int, float]` by just `float`?

Reply:
Not sure that makes sense as a general rule...